rand_distr/tests/cdf.rs

```diff
@@ -195,3 +555,434 @@ fn binomial() {
+}
+
+fn ln_binomial(n: u64, k: u64) -> f64 {
+    ln_factorial(n) - ln_factorial(k) - ln_factorial(n - k)
+}
+
+/// https://docs.rs/statrs/latest/src/statrs/function/gamma.rs.html#260-347
+fn gamma_lr(a: f64, x: f64) -> f64 {
+    let eps = 0.000000000000001;
+    let big = 4503599627370496.0;
+    let big_inv = 2.220_446_049_250_313e-16;
+
+    let ax = a * x.ln() - x - a.lgamma().0;
+    if ax < -709.782_712_893_384 {
+        if a < x {
+            return 1.0;
+        }
+        return 0.0;
+    }
+    if x <= 1.0 || x <= a {
+        let mut r2 = a;
+        let mut c2 = 1.0;
+        let mut ans2 = 1.0;
+        loop {
+            r2 += 1.0;
+            c2 *= x / r2;
+            ans2 += c2;
+
+            if c2 / ans2 <= eps {
+                break;
+            }
+        }
+        return ax.exp() * ans2 / a;
+    }
+
+    let mut y = 1.0 - a;
+    let mut z = x + y + 1.0;
+    let mut c = 0;
+
+    let mut p3 = 1.0;
+    let mut q3 = x;
+    let mut p2 = x + 1.0;
+    let mut q2 = z * x;
+    let mut ans = p2 / q2;
+
+    loop {
+        y += 1.0;
+        z += 2.0;
+        c += 1;
+        let yc = y * f64::from(c);
+
+        let p = p2 * z - p3 * yc;
+        let q = q2 * z - q3 * yc;
+
+        p3 = p2;
+        p2 = p;
+        q3 = q2;
+        q2 = q;
+
+        if p.abs() > big {
+            p3 *= big_inv;
+            p2 *= big_inv;
+            q3 *= big_inv;
+            q2 *= big_inv;
+        }
+
+        if q != 0.0 {
+            let nextans = p / q;
+            let error = ((ans - nextans) / nextans).abs();
+            ans = nextans;
+
+            if error <= eps {
+                break;
+            }
+        }
+    }
+    1.0 - ax.exp() * ans
+}
+
+fn gen_harmonic(n: u64, m: f64) -> f64 {
+    match n {
+        0 => 1.0,
+        _ => (0..n).fold(0.0, |acc, x| acc + (x as f64 + 1.0).powf(-m)),
+    }
+}
+
+/// https://docs.rs/spfunc/latest/src/spfunc/zeta.rs.html#20-43
+fn zeta_func(s: f64) -> f64 {
+    if s == 1.0 {
+        return f64::INFINITY;
+    }
+
+    let mut akn = vec![1.0];
+    let mut two_pow = 0.5;
+    let head = 1.0 / (1.0 - 2.0.powf(1.0 - s));
+    let mut tail_prev = 0.0;
+    let mut tail = two_pow * akn[0];
+
+    while (tail - tail_prev).abs() >= f64::EPSILON {
+        update_akn(&mut akn, s);
+        two_pow /= 2.0;
+        tail_prev = tail;
+        tail += two_pow * akn.iter().sum::<f64>();
+    }
+
+    head * tail
+}
+
+fn update_akn(akn: &mut Vec<f64>, s: f64) {
+    let n = akn.len() - 1;
+
+    let n1 = n as f64 + 1.0;
+
+    akn.iter_mut().enumerate().for_each(|(k, a)| {
+        let num = n1;
+        let den = n + 1 - k;
+        *a *= num / den as f64;
+    });
+    let p1 = -1.0 / n1;
+    let p2 = (n1 / (n1 + 1.0)).powf(s);
+
+    akn.push(p1 * p2 * akn[n]);
+}
+
+/// [1] FAST AND ACCURATE CALCULATION OF OWEN’S T-FUNCTION
+///     by Mike Patefield and David Tandy
+///     http://www.jstatsoft.org/v05/i05/paper
+
+fn owen_t(h: f64, a: f64) -> f64 {
```

Comment:
This appears to be CC-BY, so I think this is mostly enough. But:

- It would be nice to include the DOI (instead of the above link): https://doi.org/10.18637/jss.v005.i05
- You should clarify your modifications (ported to Rust), optionally using your name (not too important since everything is tracked in the git history anyway)